python/cloud-run-python-hello-world/app.py

```diff
@@ -0,0 +1,45 @@
+"""
+A sample Hello World server.
+"""
+import os
+import requests
+
+from flask import Flask, render_template
+
+# pylint: disable=C0103
+app = Flask(__name__)
+
+
+def get_metadata(item_name):
+    metadata_url = 'http://metadata.google.internal/computeMetadata/v1/'
+    headers = {'Metadata-Flavor': 'Google'}
+
+    try:
+        r = requests.get(metadata_url + item_name, headers=headers)
+        return r.text
+    except:
+        return 'Unavailable'
+
+
+@app.route('/')
+def hello():
+    """Return a friendly HTTP greeting."""
+    message = "Hello World"
+
+    project = get_metadata('project/project-id')
+    service = os.environ.get('K_SERVICE', 'Unknown service')
+    revision = os.environ.get('K_REVISION', 'Unknown revision')
```

Comment:
Will this work the same when running on a local minikube vs. on Cloud Run?

Reply:
TBD - don't know yet

Reply:
Note the canonical sample fallback is `???`, not `Unknown service`.
In local we expect to see `???` in the output unless something explicitly sets these env vars. Cloud Code's local emulation is expected to do something along these lines eventually if I understood correctly.